src/app/core/services/registrar.service.ts

```diff
@@ -16,4 +17,42 @@ export class RegistrarService {
       'email': email,
       'language': language});
   }
+
+  getApplicationsForVo(voId: number): Observable<Application[]> {
+    return this.apiService.post('json/registrarManager/getApplicationsForVo', {
+      'vo': voId
+    });
+  }
+
+  getApplicationsForVoWithState(voId: number, state: string[]): Observable<Application[]> {
+    if (state.length === 1) {
+      return this.apiService.post('json/registrarManager/getApplicationsForVo', {
+        'vo': voId,
+        'state': [state.pop()]
```

Comment:
```suggestion
        'state': state
```
I think this should work and it should be more general in the case of sending more states. You can then remove the if statement and use only this one call.